core-war/src/main/java/org/silverpeas/web/jobstartpage/control/JobStartPagePeasSessionController.java

```diff
@@ -199,11 +199,21 @@ public void setSubSpaceId(String subSpaceId) {
   }
 
   public void setSpaceId(String spaceUserId) {
-    if (m_NavBarMgr.setCurrentSpace(spaceUserId)) {
-      setManagedSpaceId(spaceUserId, true);
+    String spaceId = spaceUserId;
+    if (StringUtil.isDefined(spaceId)) {
+      List<SpaceInstLight> path = ServiceProvider.getService(AdminController.class).getPathToSpace(spaceId, true);
+      if (path.size() >= 2) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 2 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)